lib/ansible/cli/galaxy.py

```diff
@@ -155,7 +156,8 @@ def init_parser(self):
 
         collections_path = opt_help.argparse.ArgumentParser(add_help=False)
         collections_path.add_argument('-p', '--collection-path', dest='collections_path', type=opt_help.unfrack_path(pathsep=True),
-                                      default=C.COLLECTIONS_PATHS, action=opt_help.PrependListAction,
+                                      default=AnsibleCollectionConfig.collection_paths,
```

Comment:
Changing the default paths here is problematic because `collections_path` is also used by `execute_install`. We probably need to add these additional paths only in `execute_list_collection` and possible `execute_verify` if we want to be able to verify collections inside `site-packages` as well.

Reply:
Update for those following the PR comments: After some discussion in slack: https://github.com/ansible/ansible/issues/70147#issuecomment-646712442

So we do want it to apply to list and verify.  We want it to apply to install when install is checking whether things are already installed but do not want it to apply as a location that can be installed to.  We need to check the implementation to see if we can get away with setting the default here and letting the code sort it out or if we need to use a sentinel for the default and set it to AnsibleCollectionConfig.colection_paths as a default later in the code.

Reply:
After today's meeting, the above is the long term plan.  But for 2.10, we're going to apply to list and verify but not install.

I've now looked at the code and `-p`/`--collection-path` is implemented separately for install vs list and verify.  This PR changes the list and verify `--collection-path` but not the install `--collection-path` so it should be ready to go.

```
[pts/146@peru /srv/ansible/abadger8]$ ansible-galaxy collection list community.crypto   
[WARNING]: You are running the development version of Ansible. You should only run Ansible from "devel" if you are
modifying the Ansible engine, or trying out features under development. This is a rapidly changing source of code
and can become unstable at any point.

# /home/badger/.local/lib/python3.7/site-packages/ansible_collections
Collection       Version
---------------- -------
community.crypto 0.1.0  
[pts/146@peru /srv/ansible/abadger8]$ ansible-galaxy collection verify community.crypto
[WARNING]: You are running the development version of Ansible. You should only run Ansible from "devel" if you are
modifying the Ansible engine, or trying out features under development. This is a rapidly changing source of code
and can become unstable at any point.
[pts/146@peru /srv/ansible/abadger8]$ echo $?                           *[collection-list-site-packages]  (10:40:20)
0
[pts/146@peru /srv/ansible/abadger8]$ ansible-galaxy collection install  community.crypto
[WARNING]: You are running the development version of Ansible. You should only run Ansible from "devel" if you are
modifying the Ansible engine, or trying out features under development. This is a rapidly changing source of code
and can become unstable at any point.
Starting galaxy collection install process
Process install dependency map
Starting collection install process
Installing 'community.crypto:0.1.0' to '/home/badger/.ansible/collections/ansible_collections/community/crypto'
```

Reply:
I tested this out and it works as expected.